src/creatures/creature.cpp

```diff
@@ -26,15 +26,10 @@ Creature::Creature() {
 }
 
 Creature::~Creature() {
-	for (Creature* summon : summons) {
+	for (std::shared_ptr<Creature> summon : summons) {
 		summon->setAttackedCreature(nullptr);
 		summon->removeMaster();
 	}
-
-	for (Condition* condition : conditions) {
```

Comment:
Note: need to check if condition has a reference to the creature and make it weak_ptr

Reply:
actually, condition is stateless and this is fine :)